kedro/pipeline/node.py

```diff
@@ -168,11 +168,11 @@ def __hash__(self):
         return hash(self._unique_key)
 
     def __str__(self):
-        def _sorted_set_to_str(xset):
-            return f"[{','.join(sorted(xset))}]"
+        def _set_to_str(xset):
+            return f"[{','.join(xset)}]"
 
-        out_str = _sorted_set_to_str(self.outputs) if self._outputs else "None"
-        in_str = _sorted_set_to_str(self.inputs) if self._inputs else "None"
+        out_str = _set_to_str(self.outputs) if self._outputs else "None"
+        in_str = _set_to_str(self.inputs) if self._inputs else "None"
```

Comment:
I think these would still be sorted in the case of dicts. i.e. if `node(func, inputs={x="y", a="b"}, outputs={z="Z"})`, but that might be fine? Not sure if removing the `sorted()` will break other things.

Reply:
This is a great point. Actually it turns out the behaviour was a little weird in that it didn't reorder`inputs` but it did reorder `outputs`... From looking through some old PRs, it seems like the only reason for the ordering was to get a deterministic dictionary order in Python 3.5. But now in Python 3.6+ it's always preserved no need to keep it, so I've removed it.